### DISCUSSION TITLE
Problem with MIDI keys

### DISCUSSION THREAD

**Discussion by Gajdos1982** [GrandOrgue/grandorgue#1800] [Help]
Greetings,

I am a beginner with using MIDI keys and PC. I installed GrandOrgue, went through all the settings, connected the M-Audio Keystation 49 MK3 MIDI keys. With other programs, there is no problem with the game itself, I press a key, the tone plays, when I release it, the tone stops playing. However, with the GrandOrgue, what it does for me is that I press a key, and the note plays without stopping, as if I were holding the key down all the time. Is there a problem with the MIDI setup? As other programs, for example MuseScore, FL Studio do not have a problem with it.

**Comment by ClarkeBrunt** [comment]
Hi. Since the note starts in GO, it's clear that midi events are getting through, and the program is listening to the correct midi device.

The logical answer is that GO isn't configured to respond to the midi event being sent when you release the key.

You can check what midi events are being sent using GO Audio/Midi menu, and 'Log MIDI events' at the bottom. Events are displayed in the 'Log messages' dialog when you e.g. press keys (it says 'Warning' before each message - that's not a problem - it's just the mechanism to get them to display in the log).

When you release a key, some keyboards send a 'Note off' midi event, but others use a 'Note on' event with zero 'velocity'. I believe that GO should respond to either of these without special configuration.

I don't know how you configured the midi setup for your keyboard in the first place, but right click on a manual on the screen, to display its midi setup. Maybe read the help page for this. It is recommended to use 'Detect complex midi setup' (rather than 'Listen for event') and follow the on-screen instructions (which will be to press the lowest key, then the highest key, with minimum velocity).

I see a comment in the help saying: 'Lowest velocity must be at least 1, otherwise all note off messages will be missed' - could this be your problem?

**Reply by Gajdos1982** [reply]
Thanks for the help and answers. I tried to change the settings for quite a long time, I tried to change the "velocity" from 0 to 1, I re-initialized the MIDI.. 1x by some "miracle" I managed to get the MIDI information to be transmitted correctly and I played beautifully. But after turning off and turning on the PC again, it doesn't work again. There will probably be some deeper problem. Could it help to eliminate this problem by sending some MIDI log or screen settings? Because I try everything possible and it only worked once - probably only by "accident".

**Reply by oleg68** [reply]
Use File->Save after configuring MIDI

**Reply by ClarkeBrunt** [reply]
As @oleg68 says, do "Save" after changing things.

It does seem as though you might have the problem with Lowest velocity being set to 0, which would cause Note off events to be missed. I'm still on v3.10.1, and haven't seen the problem mentioned by @lbssousa below which might be present in later versions. If you are indeed getting Lowest velocity set to 0 after following the 'Detect complex midi setup' procedure, then this does seem to deserve further investigation.

**Comment by lbssousa** [comment]
I have experienced the same behavior with a fresh GrandOrgue 3.13.1 installation. It's closely related to issue #1762:

- When I configure my manuals/pedals using "Detect complex MIDI setup", the minimal velocity is always set to zero, not mattering how fast I press the keys, which procudes the effect of "not releasing the key". In order to avoid it, I need to manually set the minimal velocity to 1.
- When I configure some keys to toggle stops, it activates the stop when I press the key, but deactivates it when I release de key. In order to avoid that, I need to manually edit the event from "9x Note" to "9x Note On Toggle".

**Comment by ClarkeBrunt** [comment]
I hjave just done the experiment of comparing GrandOrgue versions v3.10.0-1 and v3.13.3-1 to investigate this problem with setting up manuals and toggle stops (I used the AppImage versions on Ubuntu Studio, but that probably doesn't matter).

I agree with what @lbssousa says above, and the problems didn't happen in v3.10.0-1.

@oleg68 - where is this best taken forward? The issue mentioned above #1762 got rather confusing, with discussing Launchpads and things, rather than just concentrating on setting up a simple manual or a button.

I have saved output from 'Log midi events' and can do more if needed.

To try to summarise briefly:

My keyboards send 'Note' with velocity 0 on release.

In both versions, setting up a manual with "Detect complex midi setup", if you press (and hold - important) the lowest key, wait for prompt to change, then press highest key, then the result is correct.

If however you press and release (quickly) the lowest key, wait for prompt to change, then press highest key, then in v3.10.0-1 the result is correct (i.e. the release of the lowest key appears to have been ignored). But in v3.13.3-1 'Lowest velocity' gets set to 0, causing the problem which started this thread.

Maybe you're supposed to hold the lowest key down while pressing the highest key, but I suspect many people will fail to do this.

And trying to set a phyical button (which sends 'Note' with some velocity when pressed, and 'Note' with zero velocity when released) to operate e.g. "Swell to Great" on the organ, then "Detect complex midi setup" - first you get "Please toggle it (to on state, if possible)" - do a quick press/release on the button, then "Please toggle it (to off state, if possible)" - do another quick press/release. This correctly sets "9x Note On Toggle" with v3.10.0-1, but sets "9x Note" with v3.13.3-1 (so e.g. your "Swell to Great" is only on while holding the button down, until manually edited to toggle in the dialog).

**Reply by TangoFanta** [reply]
I used to have similar issues with other midi supported programs. When mapping MIDI controls, the programs would usually map the command to the current position of the control. This meant you had to hold the control in position until the program confirmed the mapping. If you released the control too quickly, the program would map the command to when the button was released. It wasn't usually a problem, but made for an annoying delay until we figured out what was happening.
